### ISSUE TITLE
pd.concat(Iterable[pd.Series], axis=1) returns wrong type

### ISSUE THREAD

**Issue by Dr-Irv** [microsoft/python-type-stubs#69]
pyright 2021.5.4

I tried to fix this in #68, but I was fooled by my syntax error.

pyright 2021.5.4 has issue with `pd.concat` return types based on the axis argument:
```python
import pandas as pd

s1 = pd.Series([1, 2, 3])
s2 = pd.Series([4, 5, 6])

df: pd.DataFrame = pd.concat([s1, s2], axis=1)

ss1: pd.Series = pd.concat([s1,s2], axis=0)

ss2: pd.Series = pd.concat([s1,s2])

```
reports
```
Expression of type "Series[Dtype@concat]" cannot be assigned to declared type "DataFrame"
  "Series[Dtype@concat]" is incompatible with "DataFrame"
```

The `axis` argument determines return types when doing `pd.concat()` on `Series` .  Tried various fixes, but couldn't any to work for all 3 cases shown in the example.

The declaration for the first statement points to the first overload in the type stubs that corresponds to `axis=0`, but I have an explicit `axis=1`.

```python
def concat(objs: Union[Iterable[Optional[Series]], Mapping[Optional[Hashable], Optional[Series]]], join: str=..., ignore_index: bool=..., keys=..., levels=..., names=..., verify_integrity: bool=..., sort: bool=..., copy: bool=..., *, axis: Literal[0]) -> Series[Dtype]: ...
```

I don't see how the `axis=1` argument would match that overload.  I suspect there is an error in how the types are matched as opposed to being in the stubs.

**Comment by Dr-Irv** [comment]
This is still a problem with pyright 2021.6.1

**Comment by Dr-Irv** [comment]
fixed in #156 @gramster to clsoe
